honeybee_energy/constructionset.py

```diff
@@ -0,0 +1,801 @@
+
+    __slots__ = ('_exterior_construction', '_interior_construction',
+                 '_ground_construction')
+
+    def __init__(self, exterior_construction=None, interior_construction=None,
+                 ground_construction=None):
+        """Initialize set.
+
+        Args:
+            exterior_construction: An OpaqueConstruction object for faces with an
+                Outdoors boundary condition.
+            interior_construction: An OpaqueConstruction object for faces with a
+                Surface or Adiabatic boundary condition.
+            ground_construction: : An OpaqueConstruction object for faces with a
+                Ground boundary condition.
+        """
+        self.exterior_construction = exterior_construction
+        self.interior_construction = interior_construction
+        self.ground_construction = ground_construction
+
+    @property
+    def exterior_construction(self):
+        """Get or set the OpaqueConstruction for exterior Faces."""
+        return self._exterior_construction
+
+    @exterior_construction.setter
+    def exterior_construction(self, value):
+        self._exterior_construction = value
+
+    @property
+    def interior_construction(self):
+        """Get or set the OpaqueConstruction for interior Faces."""
+        return self._interior_construction
+
+    @interior_construction.setter
+    def interior_construction(self, value):
+        self._interior_construction = value
+
+    @property
+    def ground_construction(self):
+        """Get or set the OpaqueConstruction for underground Faces."""
+        return self._ground_construction
+
+    @ground_construction.setter
+    def ground_construction(self, value):
+        self._ground_construction = value
+
+    @property
+    def is_default(self):
+        """Boolean to note whether all assigned constructions are default."""
+        return self._exterior_construction is None and \
+            self._interior_construction is None and self._ground_construction is None
+
+    def to_dict(self, model_set=False):
+        """Get the Set as a dictionary.
+
+        Args:
+            model_set: A Boolean to note whether this set is being exported as
+                part of the ConstructionSet of an entire model, in which case all
+                constructions are written but individual materials are not described
+                in detail.  Otherwise, only unique constructions are written along
+                with fully detailed materials.  Default: False.
+        """
+        base = {'type': self.__class__.__name__}
+        if model_set:
+            base['exterior_construction'] = self.exterior_construction.to_dict(False)
+            base['interior_construction'] = self.interior_construction.to_dict(False)
+            base['ground_construction'] = self.ground_construction.to_dict(False)
+        else:
+            if self._exterior_construction is not None:
```

Comment:
I am going to revise the whole dictionary methods here after our discussion, given the fact that only unique materials should be written.  I can have the key present and set it to None.  I realize that it can be bad practice to just not include it at all.